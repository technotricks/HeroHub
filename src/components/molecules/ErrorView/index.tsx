import {CenteredVH, HText} from '@/components/atom';
import {SafeScreen} from '@/components/template';

const ErrorView: React.FC = () => {
  return (
    <>
      <SafeScreen>
        <CenteredVH>
          <HText color="#5b5858" aligin="center" isBold fontSize="18px">
            An error occurred. Please try again later.
          </HText>
        </CenteredVH>
      </SafeScreen>
    </>
  );
};

export default ErrorView;
