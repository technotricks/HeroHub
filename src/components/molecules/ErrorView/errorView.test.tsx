import 'react-native';
import React from 'react';

import {it, expect, jest, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {create, act} from 'react-test-renderer';
import ErrorView from '.';

describe('ErrorView component should render correctly', () => {
  const view = create(<ErrorView />);

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });
});
