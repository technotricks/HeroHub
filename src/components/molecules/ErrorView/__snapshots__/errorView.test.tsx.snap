// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView component should render correctly snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <Text
      aligin="center"
      color="#5b5858"
      fontSize="18px"
      isBold={true}
      style={
        {
          "color": "#5b5858",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      An error occurred. Please try again later.
    </Text>
  </View>
</RCTSafeAreaView>
`;
